lib/swaps/Swaps.ts

```diff
@@ -192,41 +240,33 @@ class Swaps extends EventEmitter {
    * @param taker our local taker order
    * @returns A promise that is resolved once the swap is completed, or rejects otherwise
    */
-  public executeSwap = (maker: PeerOrder, taker: OwnOrder): Promise<SwapResult> => {
-    return new Promise((resolve, reject) => {
-      if (!this.verifyExecution(maker, taker)) {
-        reject();
-        return;
-      }
+  public executeSwap = async (maker: PeerOrder, taker: OwnOrder): Promise<SwapResult> => {
+    await this.verifyExecution(maker, taker);
+    const rHash = this.beginSwap(maker, taker);
+    if (!rHash) {
+      throw new Error('cannot execute swap. rHash not found');
+    }
 
+    return await new Promise((resolve, reject) => {
       const cleanup = () => {
         this.removeListener('swap.paid', onPaid);
         this.removeListener('swap.failed', onFailed);
       };
-
       const onPaid = (swapResult: SwapResult) => {
         if (swapResult.rHash === rHash) {
           cleanup();
           resolve(swapResult);
         }
       };
-
       const onFailed = (deal: SwapDeal) => {
         if (deal.rHash === rHash) {
           cleanup();
           reject();
         }
       };
-
-      const rHash = this.beginSwap(maker, taker);
-      if (!rHash) {
-        reject();
-        return;
-      }
-
       this.on('swap.paid', onPaid);
       this.on('swap.failed', onFailed);
-    });
+    }) as SwapResult;
```

Comment:
I think it's a little bit more readable if instead of `as SwapResult` here, you did `return new Promise<SwapResult>...` on line 250. This is so far removed from the `return` statement that it's not obvious that they're part of the same line of code. Generally I prefer `as` over `<>` but it took me a second here to realize what was going on.

Reply:
Thanks. It definitely does look better. Will try to use `<>` over `as` from now on.

Reply:
`as` is definitely fine too, but I think only when it can stay on the same line as the object it's casting.